src/js/controller/Setup.js

```diff
@@ -62,11 +65,18 @@ define([
         ];
 
         this.start = function () {
+            _setupFailureTimeout = setTimeout(_setupTimeoutHandler.bind(this), _errorTimeoutDelay);
+
             _.defer(_nextTask);
         };
 
+        function _setupTimeoutHandler(){
+            this.destroy();
+            _error('Setup Timeout Error: Setup took longer than '+_errorTimeoutDelay+' seconds to complete.');
```

Comment:
I guess this is an opinion type of thing. Kyle had done it your way originally but I asked him to put the calculation into the variables declaration.

From my point of view, it is confusing to put the multiplier separate from the definition of the delay. All timeouts in javascript are measured in ms, so I don't see any reason to represent them as seconds internally. I'd rather change the message to say 'ms' , or just /1000.